### PR TITLE
Fix _after_postgeneration being invoked twice

### PR DESCRIPTION
**Pull request by youtux**
Partially fixes https://github.com/pytest-dev/pytest-factoryboy/issues/156.

### CODE REVIEW

File: pytest_factoryboy/fixture.py [hunk 1]
```diff
@@ -445,9 +468,9 @@ def subfactory_fixture(request: SubRequest, factory_class: FactoryType) -> Any:
     return request.getfixturevalue(fixture)
 
 
-def get_caller_locals(depth: int = 2) -> dict[str, Any]:
+def get_caller_locals(depth: int = 0) -> dict[str, Any]:
     """Get the local namespace of the caller frame."""
-    return sys._getframe(depth).f_locals
+    return sys._getframe(depth + 2).f_locals
```
Comment (skarzi) [inline pytest_factoryboy/fixture.py:473]: why this change is required?

File: pytest_factoryboy/fixture.py [hunk 2]
```diff
@@ -291,6 +291,24 @@ def evaluate(request: SubRequest, value: LazyFixture[T] | T) -> T:
     return value.evaluate(request) if isinstance(value, LazyFixture) else value
 
 
+def noop(*args: Any, **kwargs: Any) -> None:
+    """No-op function."""
+    pass
```
Comment (skarzi) [inline pytest_factoryboy/fixture.py:296]: NIP: `pass` is not required when docstring is provided

File: tests/test_postgen_dependencies.py [hunk 3]
```diff
@@ -179,6 +179,33 @@ def test_postgenerationmethodcall_fixture(foo: Foo):
...
+
+        @classmethod
+        def _after_postgeneration(
+            cls, obj: dict[str, Any], create: bool, results: dict[str, Any] | None = None
+        ) -> None:
+            obj.setdefault("_after_postgeneration_calls", []).append((obj, create, results))
+
+    def test_postgeneration_called_once(self, request):
+        """Test that ``_after_postgeneration`` is called only once."""
+        foo = request.getfixturevalue("collector")
```
Comment (skarzi) [inline tests/test_postgen_dependencies.py:198]: why not add `collector` as a test's function argument to let `pytest` do the job of getting the fixture value?

File: pytest_factoryboy/fixture.py (hunk 1 above)
Comment (youtux) [reply pytest_factoryboy/fixture.py:473]: not needed, but it makes more sense, so that the caller can specify if it wants the locals from the caller (from its POV) or <depth> levels from its POV

File: pytest_factoryboy/fixture.py (hunk 2 above)
Comment (youtux) [reply pytest_factoryboy/fixture.py:296]: ah yeah true that

File: tests/test_postgen_dependencies.py (hunk 3 above)
Comment (youtux) [reply tests/test_postgen_dependencies.py:198]: it's just in case there is an error in the setup, it's nicer to have the tests error out at runtime, rather than at collection time (especially if doing TDD), as it would prevent you from running other tests
